src/cmp_ccpa.js

```diff
@@ -49,10 +49,9 @@ const interactWithCMP = async (page) => {
 		.find((f) => f.url().startsWith('https://sourcepoint.theguardian.com'));
 	await frame.click('button[title="Do not sell my personal information"]');
 
-	/*
-	 As of Sep 14, some delay seems to be required before SP will persist the user's choice.
-	 */
-	await page.waitForTimeout(500);
+	await page.waitForNavigation({waitUntil: 'domcontentloaded'});
+	// We see some run failures if we do not include a wait time after a page load
+	await page.waitForTimeout(3000);
```

Comment:
I think adding `await checkCMPIsOnPage(page);` might work better here?

Reply:
We shouldn't see the CMP after the page automatically reloads so I don't think that would work here?

Reply:
Ah I see, I was confusing it with the function to check if it's visible. I'll try it out on code

Reply:
It doesn't look like this works - it times out waiting for the selector
<img width="816" alt="Screenshot 2024-03-08 at 13 28 38" src="https://github.com/guardian/commercial-canaries/assets/108270776/9944f7d0-396f-4867-9d62-e7c72fb83ba3">

Reply:
Sorry, I think you'll need to add some options to get it to find it `hidden` https://pptr.dev/api/puppeteer.page.waitforselector#remarks, any maybe increase the timeout

Reply:
Arbitrary waiting is a bit of an anti-pattern, but if you've felt like you've spent enough time on this feel free to merge!

Reply:
Would there be a problem with just using the original way of waiting that I had before? Keen to get this fixed quickly.